packages/core/admin/shared/permissions.ts

```diff
@@ -0,0 +1,52 @@
+type Condition = {
+    id: string;
+    displayName: string;
+    category: string;
+}
+
+type Permission = {
```

Comment:
I think a lot of these can be taken from the `@strapi/utils` package iirc.

Reply:
Could you be a bit more specific? 🙈 I checked it, but couldn't find anything useful ...

Reply:
sorry i wrote the wrong package – that was dumb of me:

```ts
import type { domain } from '@strapi/permissions';

type Permission = domain.permission.Permission;
```

Reply:
ah, oh ... a new world appears.

Reply:
![Joe Biden GIF by GIPHY News](https://media1.giphy.com/media/gZpVoDMMuEHRG59eJB/200.gif)

Reply:
Are we ok having `@strapi/permissions` as a dependency of `strapi/admin`?

Reply:
We're only using types, so I would assume so, this is how i've normally gone about it.

Reply:
Mh, so this is the old problem again: the types in the permission package represent the BE internal data-structure, but not what the admin app deals with:

```
export interface Permission {
  action: string;
  actionParameters?: Record<string, unknown>;
  subject?: string | object | null;
  properties?: object;
  conditions?: string[];
}
```

Reply:
@joshuaellis What do you think about leaving them here for now until the controller was converted to TS to move on? I can check with Jamie next week on how to share them best according to the ADR you wrote this week...

Tests should be fixed now.

Reply:
yes 

![Lisa Simpson Full Steam Ahead GIF by The Simpsons](https://media4.giphy.com/media/3orieOqXH7BTzuQO4w/200.gif)